### PR TITLE
Infra for regeneration of third-party-notices file

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
This is based on infra originally implemented by @dagood https://github.com/dagood/core-setup/compare/bac3b0dc8ec9e0d4d2441524ed2bca24f8bab748...100d6dc55309b01856a15bd991c784f31154f63a

It regenerates TPN file under src/installer/pkg, with content from dotnet/runtime and other repos. We used the same infra for .NET 5 release, but it was never merged into runtime repo.

### CODE REVIEW

File: eng/regenerate-third-party-notices.proj [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      <TpnRepo Include="dotnet/wpf" />
+
+      <!--
+        Additional repos that should be included but don't have any third-party-notices files:
+          dotnet/efcore
+          dotnet/extensions
+          dotnet/icu
+          dotnet/sdk
+          dotnet/windowsdesktop
+          mono/linker
```
[the remark this answers is not in the document]
Comment (NikolaMilosavljevic) [reply eng/regenerate-third-party-notices.proj]: Resolved - thanks!

**Comment by NikolaMilosavljevic** [comment]
@dagood @akoeplinger is this now good for merging? I'm waiting on one approving review.

File: eng/regenerate-third-party-notices.proj [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <TpnRepo Include="dotnet/runtime" />
+      <TpnRepo Include="dotnet/aspnetcore" />
+      <TpnRepo Include="dotnet/installer" />
+      <TpnRepo Include="dotnet/roslyn-analyzers" />
+      <TpnRepo Include="dotnet/templating" />
+      <TpnRepo Include="dotnet/winforms" />
+      <TpnRepo Include="dotnet/wpf" />
+
+      <!--
+        Additional repos that should be included but don't have any third-party-notices files:
```
Comment (akoeplinger) [inline eng/regenerate-third-party-notices.proj:25]: Do we have an issue which tracks adding these repos?

dotnet/emsdk would be another one.

**Review by akoeplinger — APPROVED** [review]

File: eng/regenerate-third-party-notices.proj (hunk 2 above)
Comment (NikolaMilosavljevic) [reply eng/regenerate-third-party-notices.proj:25]: Thanks - I've added this repo and created the tracking issue: https://github.com/dotnet/runtime/issues/61466

**Review by dagood — APPROVED** [review]

**Comment by danmoseley** [comment]
@NikolaMilosavljevic is this ok to merge now?

**Comment by danmoseley** [comment]
@NikolaMilosavljevic is this OK to merge? or else we should close it.

**Comment by NikolaMilosavljevic** [comment]
@danmoseley I'll need to verify that this still works fine with potential infra changes that happened in the meantime.

**Comment by NikolaMilosavljevic** [comment]
Verified that infra is still compatible with original changes. TPNs can be regenerated with a simple command: `build -subset RegenerateThirdPartyNotices`

**Comment by danmoseley** [comment]
Thanks @NikolaMilosavljevic !

**Pull request merged**